br/pkg/lightning/backend/local/duplicate.go

```diff
@@ -573,6 +587,99 @@ func (m *DupeDetector) RecordIndexConflictError(ctx context.Context, stream DupK
 	return nil
 }
 
+// RetrieveKeyAndValueFromErrFoundDuplicateKeys retrieves the key and value
+// from ErrFoundDuplicateKeys error.
+func RetrieveKeyAndValueFromErrFoundDuplicateKeys(err error) ([]byte, []byte, error) {
+	if !common.ErrFoundDuplicateKeys.Equal(err) {
+		return nil, nil, err
+	}
+	tErr, ok := errors.Cause(err).(*terror.Error)
+	if !ok {
+		return nil, nil, err
+	}
+	if len(tErr.Args()) != 2 {
+		return nil, nil, err
+	}
+	key, keyIsByte := tErr.Args()[0].([]byte)
+	value, valIsByte := tErr.Args()[1].([]byte)
+	if !keyIsByte || !valIsByte {
+		return nil, nil, err
+	}
+	return key, value, nil
+}
+
+// newErrFoundConflictRecords generate an error ErrFoundDataConflictRecords / ErrFoundIndexConflictRecords
+// according to key and value.
+func newErrFoundConflictRecords(key []byte, value []byte, tbl table.Table) error {
+	sessionOpts := encode.SessionOptions{
+		SQLMode: mysql.ModeStrictAllTables,
```

Comment:
Conflict detection related code like the following all use `mysql.ModeStrictAllTables`, I think it would be ok if here we also use `mysql.ModeStrictAllTables`?
https://github.com/pingcap/tidb/blob/0d742d35f80eb9178105b1a09ee41eb693e6dbb4/br/pkg/lightning/importer/table_import.go#L1023-L1027
https://github.com/pingcap/tidb/blob/0d742d35f80eb9178105b1a09ee41eb693e6dbb4/pkg/ddl/ingest/backend.go#L98-L102